### PR TITLE
Sort

### PR DESCRIPTION
**Pull request by dave-hickman**

### CODE REVIEW

File: src/pages/Articles.jsx [hunk 1]
```diff
@@ -1,28 +1,49 @@
 import { useEffect, useState } from "react";
+import {useSearchParams} from "react-router-dom"
 import ArticleSummary from "../components/ArticleSummary";
 import Sort from "../components/Sort";
 import { getArticles } from "../utils";
 import "../Articles.css"
 
-const Articles = () => {
+const Articles = ({setQueryString, queryString}) => {
```
Comment (JVal-NC) [inline src/pages/Articles.jsx]: Does queryString state need to be in App?

Theres also a bug with comment count sorting. May be useful to show the votes and comment counts on the review cards if we are allowing the user to sort by these to provide a sense of confidence.

File: src/pages/Articles.jsx [hunk 2]
```diff
@@ -1,28 +1,49 @@
...
+      setQueryString(`?order=${ascDesc}`)
+    }
+    else if (sort && !ascDesc){
+      setQueryString(`?sort_by=${sort}`)
+    }
+    else setQueryString("")
+  }
+
 
   useEffect(() => {
```
Comment (JVal-NC) [inline src/pages/Articles.jsx:24]: We may want to set loading to true inside the useEffect if doing another api request.

File: src/utils.js [hunk 3]
```diff
@@ -4,8 +4,9 @@ const news = axios.create({
   baseURL: "https://news-project-g3uj.onrender.com/api",
 });
 
-const getArticles = async () => {
-  const { data } = await news.get("/articles");
+const getArticles = async (queryString) => {
```
Comment (JVal-NC) [inline src/utils.js:7]: Could we rename this file to api, it is a util but api would be more specific to what the functions purpose are.

**Review by JVal-NC — COMMENTED** [review]
Good work so far, codes looking good. A bug to fix and some state to possibly relocate is all.

File: src/components/Sort.jsx [hunk 4]
```diff
@@ -1,5 +1,27 @@
...
+
+  const handleOrderChange = (e) => {
+    setAscDesc(e.target.value)
+  }
+
+  return (
+    <section>
+      <label htmlFor="sort">Sort by:</label>
+    <select onChange={handleSortChange}>
+      <option value="">Date</option>
```
Comment (Augs0) [inline src/components/Sort.jsx]: This option is missing a value

File: src/components/Sort.jsx [hunk 5]
```diff
@@ -1,5 +1,27 @@
...
+      <option value="comment_count">Comment Count</option>
+      <option value="votes">Votes</option>
+    </select>
+    <select onChange={handleOrderChange}>
+      <option value="">Descending</option>
```
Comment (Augs0) [inline src/components/Sort.jsx]: And this one :)

File: src/pages/Articles.jsx [hunk 6]
```diff
@@ -23,7 +23,9 @@ const Articles = () => {
     } else setQueryString("");
   };
 
+  
   useEffect(() => {
+    setSearchParams(queryString);
     const fetchArticles = async () => {
       const response = await getArticles(queryString);
```
Comment (Augs0) [inline src/pages/Articles.jsx:28]: You should set your loading to false by default , then true in here, then finally false again to better represent what is happening

File: src/pages/IndividualArticle.jsx [hunk 7]
```diff
@@ -45,14 +45,24 @@ const IndividualArticle = ({userId}) => {
         <div className="author-and-votes-container">
           <h3 className="author">{article.article[0].author}</h3>
           <div className="votes-container">
-            <button aria-label="Upvote this post" className="vote-button vote-up" onClick={() => handleVote(1)}>
+            <button
+              aria-label="Upvote this post"
```
Comment (Augs0) [inline src/pages/IndividualArticle.jsx:49]: Yay! Nice, accessible buttons 🍪

**Review by Augs0 — CHANGES_REQUESTED** [review]
Some really good work here Dave :D you handled all the feedback from Joe, but I'd like you to also implement sorting on articles by topic. There's a few other comments I've added here to for extra points

**Review by Augs0 — CHANGES_REQUESTED** [review]
Hey, this is looking great and I love the topic link on the cards! It's almost there but the thing I was asking for was to include a sort dropdown on your topics page. For example, if I look at the cooking articles, I want to sort those by date, comment, count etc :)

**Review by Augs0 — APPROVED** [review]
We got there in the end :D and like I said, what you added previously was still a nice touch so nothing lost ✨  you're good to merge!

**Pull request merged**